coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -4525,6 +4549,25 @@ def arange(context, node):
     context.add(res)
 
 
+@register_torch_op(torch_alias=["arange.start_step"])
+def arange_start_step(context, node):
+    inputs = _get_inputs(context, node)
+    start = inputs[0]
+    end = inputs[1]
+    step = 1 if len(inputs) < 3 else inputs[2]
+
+    int_start = isinstance(start, int) or types.is_int(start.dtype)
+    int_end = isinstance(end, int) or types.is_int(end.dtype)
+    int_step = isinstance(step, int) or types.is_int(step.dtype)
+
+    if int_start != int_end or int_start != int_step:
+        start = mb.cast(x=start, dtype="fp32")
+        end = mb.cast(x=end, dtype="fp32")
+        step = mb.cast(x=step, dtype="fp32")
```

Comment:
I think you could just use the `promote_input_dtypes` helper function and cut the size of this function in half.

Reply:
Revised as suggested